astropy/units/tests/test_quantity.py

```diff
@@ -830,14 +830,6 @@ def __quantity_instance__(self, *args, **kwargs):
     assert_allclose(q4.value, q5.value)
 
 
-# Regression test for issue #2315, divide-by-zero error when examining 0*unit
-
-
-def test_self_equivalency():
-    assert u.deg.is_equivalent(0 * u.radian)
```

Comment:
Our API doesn't promise that a `Quantity` input works, so it's fine if it doesn't: https://github.com/astropy/astropy/blob/f7dc2aaa27fe8f43d1dae3e277d5d7a05e74d35b/astropy/units/core.py#L953-L959

Reply:
The `assert` with `(1 * u.rad)` still works, so it would be possible to keep it.

The reason this test didn't fail in #17048 was that `_UnitMetaClass.__call__()` allows converting `Quantity` instances to `CompositeUnit` without sanitizing the scale at all, which is why I had to add brand new `sanitize_scale()` calls to that method in this pull request.

Reply:
Ah, so this is a bug fix too! Thanks for finding that.

Reply:
This is an example of a bug that type checkers could find very easily in a well annotated codebase.